Code/app/src/main/java/com/cmpt362/fitcheck/firebase/Firebase.kt

```diff
@@ -98,4 +103,29 @@ object Firebase {
 
         return null
     }
+
+    // function for getting all users from the database with a view Model
+    fun loadAllUsers(allUsers: MutableLiveData<List<User>>) {
+        usersReference.addValueEventListener(object : ValueEventListener{
+            override fun onDataChange(snapshot: DataSnapshot) {
+                try {
+
+                    val userList : List<User> = snapshot.children.map { dataSnapshot ->
+                        println("debug: dataSnapshot - $dataSnapshot")
+                        dataSnapshot.getValue(User::class.java)!!
+
+                    }
+                    println("debug: userList - $userList")
+                    allUsers.postValue(userList)
+                } catch (e: Exception){
+                    println("debug: Exception when loading users $e")
+                }
+            }
+
+            override fun onCancelled(error: DatabaseError) {
+//                TODO("Not yet implemented")
+                println("debug: onCalled when loading Users $error")
```

Comment:
I think you meant to write onCancelled? 
Also we should probably remove the TODO comment